plugins/outputs/cloud_pubsub/cloud_pubsub.go

```diff
@@ -64,9 +66,16 @@ func (ps *PubSub) Connect() error {
 		return fmt.Errorf(`"project" is required`)
 	}
 
+	var err error
+	ps.encoder, err = internal.NewContentEncoder(ps.ContentEncoding)
```

Comment:
As above, I'd prefer during init you check the value of content encoding and verify it is one of identity or gzip